projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-transient.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * REST API endpoint for editing Jetpack Transients.
+ *
+ * @package automattic/jetpack
+ * @since 9.7.0
+ */
+
+/**
+ * Jetpack transients API.
+ *
+ * @since 9.7.0
+ */
+class WPCOM_REST_API_V2_Endpoint_Transient extends WP_REST_Controller {
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->namespace = 'wpcom/v2';
+		$this->rest_base = 'transients';
+		add_action( 'rest_api_init', array( $this, 'register_routes' ) );
+	}
+
+	/**
+	 * Called automatically on `rest_api_init()`.
+	 */
+	public function register_routes() {
+		// DELETE /sites/<blog-id>/transients/$name route.
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<name>\w{1,172})',
+			array(
+				array(
+					'methods'             => WP_REST_Server::DELETABLE,
+					'callback'            => array( $this, 'delete_transient' ),
+					'permission_callback' => array( $this, 'delete_transient_permissions_check' ),
+					'args'                => array(
+						'name' => array(
+							'description'       => __( 'The name of the transient to delete.', 'jetpack' ),
+							'required'          => true,
+							'type'              => 'string',
+							'sanitize_callback' => 'sanitize_text_field',
+						),
+					),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Delete transient callback.
+	 *
+	 * @param \WP_REST_Request $request Full details about the request.
+	 * @return array|WP_Error
+	 */
+	public function delete_transient( \WP_REST_Request $request ) {
+		$transient_name = $request->get_param( 'name' );
+		if ( false !== strpos( $transient_name, 'jetpack_connected_user_data_' ) &&
+			wp_get_current_user()->ID === (int) substr( $transient_name, 28 ) ) {
```

Comment:
Do we need to worry about use-cases where `ID` isn't set?

Reply:
According to [the doc](https://developer.wordpress.org/reference/classes/wp_user/), I think it worths checking whether the current user exists:

Check if the user exists:

```php
<?php
    $current_user = wp_get_current_user();
    if ( $current_user->exists() ) {
       // do something
    }    
?>
```

Even better I think, we would use [get_current_user_id()](https://developer.wordpress.org/reference/functions/get_current_user_id/) instead:

> Return #Return
> (int) The current user's ID, or 0 if no user is logged in.

```suggestion
			! empty( get_current_user_id() ) &&
			get_current_user_id() === (int) substr( $transient_name, 28 ) ) {
```

Something like this?

Reply:
While testing I realize it's going to delete the transient only when its name tidies with the current user. I think it won't fit very well with the async job process that we're trying to implement.

> To delete the jetpack_connected_user_data_$user_id transient, we should probably check for something like current_user_can( 'edit_user', $user_id )...

Reply:
> Do we need to worry about use-cases where ID isn't set?

We shouldn't need to, since there's a `permission_callback` that checks for a user [with the `read` capability](https://wordpress.org/support/article/roles-and-capabilities/#read) before running `delete_transient()`.

> Even better I think, we would use get_current_user_id() instead:

Thanks, I changed to `get_current_user_id()`: https://github.com/Automattic/jetpack/pull/19632/commits/7a7987c591d8cdc4dcaac48a55ce054f0801e8e9

> While testing I realize it's going to delete the transient only when its name tidies with the current user. I think it won't fit very well with the async job process that we're trying to implement.

From @pablinos' research into how we will call this endpoint, it sounds like we will be able to get the local user ID from the Jetpack site in the async job.

Reply:
> From @pablinos' research into how we will call this endpoint, it sounds like we will be able to get the local user ID from the Jetpack site in the async job.

Excellent